scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts

```diff
@@ -260,40 +267,36 @@ export const generateRelatedAssetIndex = async () => {
   // remove relatedAssetKey from the existing data to ensure the related assets get updated
   Object.values(assetDataWithRelatedAssetKeys).forEach(asset => delete asset.relatedAssetKey)
 
-  const progressBar = new SingleBar({}, Presets.shades_classic)
-  progressBar.start(Object.keys(generatedAssetData).length, 0)
-
-  const throttle = createThrottle({
+  const { throttle, clear: clearThrottleInterval } = createThrottle({
     capacity: 50, // Reduced initial capacity to allow for a burst but not too high
     costPerReq: 1, // Keeping the cost per request as 1 for simplicity
     drainPerInterval: 25, // Adjusted drain rate to replenish at a sustainable pace
     intervalMs: 2000,
   })
-  let i = 0
   for (const batch of chunkArray(Object.keys(generatedAssetData), BATCH_SIZE)) {
     await Promise.all(
       batch.map(async assetId => {
         await processRelatedAssetIds(assetId, assetDataWithRelatedAssetKeys, relatedAssetIndex)
         await throttle()
+        return
       }),
     )
-    i += BATCH_SIZE
-    progressBar.update(i)
   }
 
-  progressBar.stop()
+  clearThrottleInterval()
```

Comment:
Clear the interval to ensure no handles stay open forever resulting in `process.exit()` not actually exiting the process